iothub/device/samples/convention-based-samples/Thermostat/ThermostatSample.cs

```diff
@@ -74,13 +74,40 @@ private async Task HandlePropertyUpdatesAsync(ClientPropertyCollection writableP
                 switch (writableProperty.Key)
                 {
                     case "targetTemperature":
-                        const string tagetTemperatureProperty = "targetTemperature";
+                        const string targetTemperatureProperty = "targetTemperature";
                         double targetTemperatureRequested = Convert.ToDouble(writableProperty.Value);
-                        _logger.LogDebug($"Property: Received - [ \"{tagetTemperatureProperty}\": {targetTemperatureRequested}°C ].");
+
+                        _logger.LogDebug($"Property: Received - [ \"{targetTemperatureProperty}\": {targetTemperatureRequested}°C ].");
+
+                        // Evaluate if this is a request that needs to be responded to, i.e.:
+                        // -> The twin's last reported "targetTemperature" value is not the same as the requested value.
+                        // -> The twin's last reported "targetTemperature" version is less than the requested property version.
+
+                        // Retrieve the device's properties.
+                        ClientProperties properties = await _deviceClient.GetClientPropertiesAsync();
```

Comment:
I'd rather see the device get properties when it connects, and hold them in memory. Then it can compare its latest value (and version) to requests that come in.

Reply:
Interesting; since the twin value will keep on changing, are you suggesting that we prepend each twin operation with a call to GetClientProperties?

Reply:
Fetching the twin on initial startup is something that docs talk about as well, so it would be good for us to add it into our sample.  I wonder if there is any value in fetching it throughout the lifetime of the sample, since I worry that it might misguide customers into thinking that this is something that they have to do always (extra API call = more bandwidth consumed + more expensive)

Reply:
> since the twin value will keep on changing, are you suggesting that we prepend each twin operation with a call to GetClientProperties?

No. The desired value will change and the client (while connected) will be notified. That means it only needs to know what the latest request was when it connects. When it first connects it should find out what the latest reported and desired values are. It'll initialize it's local, in-memory copy to the latest reported value (and store the version for later version checking), evaluate any new desired values, and then wait for future desired values.

Reply:
I don't see any value in getting the latest reported value during a property update callback.